File: Abstract Factory/Ruby/TeamPromoConnectionFactory.rb
require_relative "IConnectionFactory.rb"
require_relative "TeamPromoHTTPConnection.rb"
require_relative "TeamPromoRPCConnection.rb"

class TeamPromoConnectionFactory < IConnectionFactory
    def createHTTPConnection
        TeamPromoHTTPConnection.new
    end

    def createRPCConnection
        TeamPromoRPCConnection.new
    end
end

File: Abstract Factory/Javascript/TeamPromoConnectionFactory.js
import IConnectionFactory from "./IConnectionFactory.js";
import TeamPromoHTTPConnection from "./TeamPromoHTTPConnection.js";
import TeamPromoRPCConnection from "./TeamPromoRPCConnection.js";

class TeamPromoConnectionFactory extends IConnectionFactory {
  createRPCConnection() {
    return new TeamPromoRPCConnection();
  }

  createHTTPConnection() {
    return new TeamPromoHTTPConnection();
  }
}

export default TeamPromoConnectionFactory;